ir/node.cpp

```diff
@@ -94,4 +102,24 @@ Util::JsonObject* IR::Node::sourceInfoJsonObj() const {
     return json;
 }
 
+void IR::Node::sourceInfoToJSON(JSONGenerator &json) const {
+    Util::SourceInfo si = srcInfo;
+    unsigned lineNumber, columnNumber;
+    cstring fName = prepareSourceInfoForJSON(si, &lineNumber, &columnNumber);
```

Comment:
Refactoring it to an `operator<<` makes sense.

As for including the `end` coordinates, I didn't do it for the following reasons:

- The main reason for including `srcInfo` in the output is to help with error messages in post processing tools. The common practice in UNIX tools is to specify something in the format `filename:line` or, for extra credit, `filename:line:column`. Ranges are very rarely used and AFAIK there isn't even a standard format for specifying them in a way that IDEs etc. will understand.

- The  `sourceInfoJsonObj` also only printed the start position, and I wanted to be consistent with it. As a bonus, it includes a `source_fragment`, which would be useful in printing an improved error message without having to include an end position. 

I suggest that these are sufficient reasons for avoiding including an explicit end position in the output. That said, if you feel strongly that both should be included, should it also be done for the `sourceInfoJsonObj`, then?

Idea: We could emit the unadjusted position, and only used the adjusted position for computing the `source_fragment` - and use this combination in both `sourceInfoToJSON` and `sourceInfoJsonObj`. Would that be acceptable?